library/objective-c/Envoy.mm

```diff
@@ -0,0 +1,45 @@
+#import "library/objective-c/Envoy.h"
+
+#import "library/common/main_interface.h"
+
+
+@interface Envoy ()
+@property (nonatomic, strong) NSThread *runner;
+@end
+
+@implementation Envoy
+
+@synthesize runner;
+
+// Create a new Envoy instance. The Envoy runner NSThread is started as part of instance
+// initialization with the configuration provided.
+- (id)initWithConfig:(NSString *)config {
+  self.runner = [[NSThread alloc] initWithTarget:self selector:@selector(run:) object:config];
+  [self.runner start];
+}
+
+// Returns whether this Envoy instance is currently active and running.
+- (BOOL)isRunning {
+  return self.runner != NULL && self.runner.isExecuting;
```

Comment:
I don't think these `!= NULL` checks are required, since this should return `FALSE` if it doesn't exist

Reply:
That's right, I had forgotten about objective-c's safety/fanciness there. Will update.